### PR TITLE
fix: tolerate wrongly-typed annotations

### LINKED ISSUE #29149: [BUG] White page when opening test steps in html report
### System info
- Playwright Version: [v1.41]
- Operating System: [macOS 14.2.1]
- Browser: [Chromium, Firefox, I guess all]
- Other info:

### Source code
[playwright-report.zip](https://github.com/microsoft/playwright/files/14038453/playwright-report.zip)

**Steps**
Open the attached report - it shows all tests as expected: 
image

However, when you try to open the tests steps of any of those tests, you'll get a blank page.
image

In console, there is error: 
`Uncaught (in promise) Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7B%7D for the full message or use the non-minified dev environment for full errors and additional helpful warnings.`

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/29149

### CODE REVIEW

File: packages/html-reporter/tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "useDefineForClassFields": true,
     "lib": ["DOM", "DOM.Iterable", "ESNext"],
     "allowJs": true,
-    "skipLibCheck": false,
```
Comment (yury-s) [inline packages/html-reporter/tsconfig.json:7]: why is this necessary?

**Review by yury-s — APPROVED** [review]

File: packages/html-reporter/tsconfig.json (hunk 1 above)
Comment (pavelfeldman) [reply packages/html-reporter/tsconfig.json:7]: Otherwise npm run build fails on tsc
